drivers/scmi-msg/scmi_pinctrl.h

```diff
@@ -0,0 +1,197 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2023, EPAM Systems
+ */
+#ifndef SCMI_MSG_PINCTRL_H
+#define SCMI_MSG_PINCTRL_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <lib/utils_def.h>
+
+#define SCMI_PROTOCOL_VERSION_PINCTRL 0x10000U
+
+#define SCMI_PINCTRL_ATTR_NAME_SZ 16U
+#define SCMI_PINCTRL_NAME_MAX_SZ  64U
+
+#define ALIGN_NEXT(VALUE, INTERVAL)                                            \
+	((((VALUE) + (INTERVAL)-1) / (INTERVAL)) * (INTERVAL))
+
+/*
+ * Identifiers of the SCMI Pinctrl Domain Management Protocol commands
+ */
+enum scmi_pinctrl_domain_command_id {
+	SCMI_PINCTRL_ATTRIBUTES = 0x03,
+	SCMI_PINCTRL_LIST_ASSOCIATIONS = 0x04,
+	SCMI_PINCTRL_SETTINGS_GET = 0x05,
+	SCMI_PINCTRL_SETTINGS_CONFIGURE = 0x6,
+	SCMI_PINCTRL_REQUEST = 0x7,
+	SCMI_PINCTRL_RELEASE = 0x8,
+	SCMI_PINCTRL_NAME_GET = 0x9,
+	SCMI_PINCTRL_SET_PERMISSIONS = 0xa,
+};
+
+#define FLD(mask, val)     (((val) << (__builtin_ffsll(mask) - 1) & (mask)))
```

Comment:
Are you sure that this builtin is available in all supported compilers? There is `GENMASK_` macros in `utils_def.h`

Reply:
reworked/removed